### PR TITLE
Move the error unwrapper into common

### PR DESCRIPTION
**Pull request by forgetso**

### CODE REVIEW

File: dev/config/src/vite/vite.test.config.ts [hunk 1]
```diff
@@ -52,6 +53,7 @@ export default function () {
 			},
 			pool: "forks", // forks is slower than 'threads' but more compatible with low-level libs (e.g. bcrypt)
 			testTimeout: 10000,
+			//disableConsoleIntercept: true,
```
Comment (HughParry) [inline dev/config/src/vite/vite.test.config.ts]: Remove new unused comment

File: packages/common/src/error.ts [hunk 2]
```diff
@@ -176,3 +181,42 @@ export class ProsopoApiError extends ProsopoBaseError<ApiContextParams> {
...
+		} else {
+			jsonError.message = JSON.parse(err.message);
+		}
+	}
+
+	jsonError.code = jsonError.code || code;
+	return { code, statusMessage, jsonError };
+};
+
+export function isZodError(err: unknown): err is ZodError {
```
Comment (HughParry) [inline packages/common/src/error.ts]: using consts or functions? different declaration in same file between line 184 and line 218

File: packages/common/src/error.ts [hunk 3]
```diff
@@ -176,3 +181,42 @@ export class ProsopoApiError extends ProsopoBaseError<ApiContextParams> {
 		this.code = code;
 	}
 }
+
+export const unwrapError = (err: ProsopoApiError | SyntaxError | ZodError) => {
```
Comment (HughParry) [inline packages/common/src/error.ts]: Why is this typed as a ProsopoApiError? Aren't there other error types which we might want to unwrap?

**Review by HughParry — APPROVED** [review]

File: packages/common/src/error.ts (hunk 2 above)
Comment (forgetso) [reply packages/common/src/error.ts]: I've changed it to a const.

Don't understand the second comment. What do you mean about a different declaration?

File: packages/common/src/error.ts (hunk 3 above)
Comment (forgetso) [reply packages/common/src/error.ts]: At the moment its only being used for ProsopoApiError, but yes it should probably be ProsopoBaseError. Changed.

File: packages/common/src/error.ts (hunk 2 above)
Comment (HughParry) [reply packages/common/src/error.ts]: was trying to communicate:

Don't mind function or const, both are fine. Just imo best to keep consistent within files, not use both

**Pull request merged**